src/native/managed/compile-native.proj

```diff
@@ -0,0 +1,67 @@
+<Project Sdk="Microsoft.Build.NoTargets">
+    <PropertyGroup>
+        <!-- We always want to use release for publishing using NativeAOT -->
+        <NativeLibsPublishConfiguration>Release</NativeLibsPublishConfiguration>
+	<!-- we always want to make shared libs -->
+        <NativeLibKind Condition="'$(NativeLibKind)' == ''">shared</NativeLibKind>
+    </PropertyGroup>
```

Comment:
I think we can simplify this if we make it a traversal project.

```suggestion
<Project Sdk="Microsoft.Build.Traversal" DefaultTargets="Publish">
    <PropertyGroup>
        <!-- We always want to use release for publishing using NativeAOT -->
        <NativeLibsPublishConfiguration>Release</NativeLibsPublishConfiguration>
	<!-- we always want to make shared libs -->
        <NativeLibKind Condition="'$(NativeLibKind)' == ''">shared</NativeLibKind>
        <!-- When we publish, we want to ensure the SDK does the same thing as though we ran 'dotnet publish' -->
       <TraversalPublishGlobalProperties>$(TraversalPublishGlobalProperties);_IsPublishing=true</TraversalPublishGlobalProperties>
    </PropertyGroup>
```

Reply:
done.  thanks, this was a great idea!